helper/timeutil/timeutil.go

```diff
@@ -142,3 +142,24 @@ func SkipAtEndOfMonth(t *testing.T) {
 		t.Skip("too close to end of month")
 	}
 }
+
+// This interface allows unit tests to substitute in a simulated Clock.
+type Clock interface {
+	Now() time.Time
+	NewTicker(time.Duration) *time.Ticker
+	NewTimer(time.Duration) *time.Timer
+}
+
+type DefaultClock struct{}
```

Comment:
Maybe good to enforce `DefaultClock` to implement `Clock`? Like adding this line:
```
_ Clock       = (*DefaultClock)(nil)
```